subprojects/internal-testing/src/main/groovy/org/gradle/testing/internal/util/Specification.groovy

```diff
@@ -18,36 +18,16 @@ package org.gradle.testing.internal.util
 
 import groovy.transform.CompileStatic
 import groovy.transform.stc.ClosureParams
-import groovy.transform.stc.FirstParam
 import groovy.transform.stc.SecondParam
 
 /**
- * This fixes Spock's with() methods to be understandable by IDEA.
- * By using this subclass, IDEA knows the closure delegate type when using with().
- * This means that you get editing assistance in said blocks WRT the object under assertion.
+ * This fixes Spock's Mock() and Stub() methods to be understandable by IDEA.
+ * By using this subclass, IDEA knows the closure delegate type when using Mock() or Stub().
+ * This means that you get editing assistance in said blocks WRT the type being mocked/stubbed.
  */
 @CompileStatic
 class Specification extends spock.lang.Specification {
 
-    @Override
```

Comment:
@lptr FYI, because I saw you had a similar change for Groovy 2.5

Reply:
Actually, it's the other two methods that Groovy had a problem with...

Reply:
```text
> Task :internalTesting:compileGroovy
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
startup failed:
/Users/lptr/Workspace/gradle/gradle/subprojects/internal-testing/src/main/groovy/org/gradle/testing/internal/util/Specification.groovy: 39: [Static type checking] - Cannot find matching method spock.lang.Specification#MockImpl(<unknown parameter type>, <unknown parameter type>, java.lang.Class <T>, groovy.lang.Closure). Please check if the declared type is correct and if the method exists.
 @ line 39, column 9.
           super.Mock(type, closure)
           ^

/Users/lptr/Workspace/gradle/gradle/subprojects/internal-testing/src/main/groovy/org/gradle/testing/internal/util/Specification.groovy: 49: [Static type checking] - Cannot find matching method spock.lang.Specification#StubImpl(<unknown parameter type>, <unknown parameter type>, java.lang.Class <T>, groovy.lang.Closure). Please check if the declared type is correct and if the method exists.
 @ line 49, column 9.
           super.Stub(type, closure)
           ^

2 errors
```